src/operations/apply.rs

```diff
@@ -56,6 +56,12 @@ pub fn apply(
         ));
     }
 
+    // Create a temporary data directory
+    let staging_data_dir = tempfile::Builder::new()
+        .prefix("artifacts-")
+        .tempdir_in(data_path)?;
```

Comment:
I haven't used this lib before so just ignore this if it doesn't make sense: Why not use `Builder::new().tempdir()` since it's creating a dir here?

Reply:
Are you suggesting that temp dir can just go to wherever they go by default? (According to docs, `std::env::temp_dir()`)

Reply:
The library creates the temp dir somewhere else by default. Specifying the data path as parent path seems like a good idea to isolate the operations under one dir.